apps/extension/src/ui/apps/popup/components/TotalFiatBalance.tsx

```diff
@@ -1,111 +1,151 @@
+import { AnalyticsEventName, AnalyticsPage, sendAnalyticsEvent } from "@ui/api/analytics"
+import { Fiat } from "@ui/domains/Asset/Fiat"
+import { useCopyAddressModal } from "@ui/domains/CopyAddress"
 import { useAnalytics } from "@ui/hooks/useAnalytics"
 import useBalances from "@ui/hooks/useBalances"
+import { useIsFeatureEnabled } from "@ui/hooks/useFeatures"
 import { useSetting } from "@ui/hooks/useSettings"
-import { FC, useCallback } from "react"
+import { ComponentProps, MouseEventHandler, useCallback, useMemo } from "react"
 import { useTranslation } from "react-i18next"
-import styled from "styled-components"
+import { PillButton } from "talisman-ui"
 
-const ToggleHide = styled(IconButton).attrs((props: { showIcon?: boolean }) => ({
-  showIcon: !!props.showIcon,
-}))`
-  position: absolute;
-  top: 0;
-  left: 0.5rem;
-  display: ${({ showIcon }) => (showIcon ? "inline-block" : "none")};
-  font-size: 1.6rem;
-  width: 2rem;
-  height: 2rem;
-`
-const Container = styled.div`
-  display: flex;
-  flex-direction: column;
-  align-items: center;
-  justify-content: center;
-
-  span {
-    font-size: var(--font-size-large);
-    font-weight: var(--font-weight-bold);
-  }
-
-  &:hover {
-    ${ToggleHide} {
-      display: inline-block;
-    }
-  }
-
-  .fiat {
-    width: 65%;
-    text-align: center;
-  }
+type Props = {
+  className?: string
+  mouseOver: boolean
+}
 
-  .balance-revealable {
-    border-radius: 10px;
-  }
-  .balance-revealable::after {
-    border-radius: 10px;
-  }
-`
+export const TotalFiatBalance = ({ className, mouseOver }: Props) => {
+  const { t } = useTranslation()
 
-const TitleRow = styled.div`
-  width: 100%;
-  display: flex;
-  justify-content: center;
-  margin-bottom: 1rem;
-  align-items: center;
-`
+  const balances = useBalances("portfolio")
 
-const Title = styled.h3`
-  font-size: var(--font-size-small);
-  color: var(--color-mid);
-  margin: 0;
-`
+  const [hideBalances, setHideBalances] = useSetting("hideBalances")
+  const { genericEvent } = useAnalytics()
 
-const Side = styled.div`
-  position: relative;
-  font-size: 1.4rem;
-  flex-grow: 0.5;
-  height: 2rem;
-`
+  const toggleHideBalance: MouseEventHandler<HTMLButtonElement> = useCallback(
+    (event) => {
+      event.stopPropagation()
+      genericEvent("toggle hide balance")
+      setHideBalances((prev) => !prev)
+    },
+    [genericEvent, setHideBalances]
+  )
 
-// force height to prevent flickering
-const TotalFiat = styled.div`
-  height: 2.9rem;
+  return (
+    <div className={classNames(className, "flex w-full items-center")}>
+      <div className="flex flex-grow flex-col items-start gap-4">
+        <div className="text-body-secondary mt-2 flex gap-2 text-sm">
+          <span>{t("Total Portfolio")}</span>
+          <button
+            className={classNames(
+              "hover:text-body focus:text-body opacity-0 transition-[color,opacity]",
+              (hideBalances || mouseOver) && "opacity-100"
```

Comment:
tailwind can easily have unintended behavior if multiple rules are overriding each other because it's the order of rule declaration in the css file that matters.
unless we integrate https://www.npmjs.com/package/tailwind-merge it's best to use this pattern
```suggestion
              "hover:text-body focus:text-body transition-[color,opacity]",
              hideBalances || mouseOver ? "opacity-100" : "opacity-0"
``` 😄

Reply:
note : there seem to be unwanted characters at the end of the suggestion :smile: , don't commit as-is.